ci: Setup GHA

mityu opened the pull request:
Run
- deno check
- deno lint
- deno fmt --check
on GHA.

mityu committed:
ci: Setup GHA

4513ECHO commented on .github/workflows/ci.yml:
```diff
@@ -0,0 +1,19 @@
+        with:
+          deno-version: v1.x
+      - name: Check types
+        run: shopt -s globstar; deno check denops/**/*.ts
```
Running `deno check` with specific directory is supported in latest Deno.
```suggestion
        run: deno check denops
```

mityu replied:
Thank you.

mityu committed:
ci: Pass directory name to deno check command

Deno 1.46.0 or later have support for running deno check command in style of "deno check <directory>"

Co-authored-by: Hibiki <mail@4513echo.dev>

kuuote commented:
Thanks.
